### PR TITLE
Updated BentoML installation documentation and example

### PR DESCRIPTION
**Pull request by ssheng**
**Description of your changes:**

- Updated the BentoML installation and deployment documentation to include both deployment options, using `BentoRequest` and `Bento` CRDs.
- Updated the BentoML example to align with the example provided from the BentoML GitHub page.
- Updated Kubeflow manifest README to include installation instructions for BentoML components.

**Checklist:**
- [ ] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`

### CODE REVIEW

**Comment by juliusvonkohout** [comment]
@ssheng you need to sign the CLA

File: contrib/bentoml/README.md
Comment (juliusvonkohout) [inline contrib/bentoml/README.md:46]: Please name it Workbenches as that will be the case in 1.8

File: contrib/bentoml/README.md
Comment (juliusvonkohout) [inline contrib/bentoml/README.md:42]: Please include the images in the repository itself.

File: contrib/bentoml/README.md
Comment (juliusvonkohout) [inline contrib/bentoml/README.md:23]: Please include the images in the repository itself.

File: contrib/bentoml/README.md
Comment (juliusvonkohout) [inline contrib/bentoml/README.md:102]: Please include the images in the repository itself.

File: README.md
Comment (juliusvonkohout) [inline README.md:300]: Kubeflow Workbenches is the name for 1.8 not Notebooks

File: contrib/bentoml/README.md
Comment (juliusvonkohout) [inline contrib/bentoml/README.md:3]: Kubeflow Workbenches is the name for 1.8 not Notebooks

File: contrib/bentoml/README.md
Comment (ssheng) [reply contrib/bentoml/README.md:23]: Removed this image and moved the detailed example to the BentoML repo. The documentation here will link the full example in BentoML.

File: contrib/bentoml/README.md
Comment (ssheng) [reply contrib/bentoml/README.md:102]: Removed this image and moved the detailed example to the BentoML repo. The documentation here will link the full example in BentoML.

File: contrib/bentoml/README.md
Comment (ssheng) [reply contrib/bentoml/README.md:113]: Added a warning.

> WARNING: The `yatai-image-builder` operator requires root privileges because it needs to access the Docker daemon, which requires elevated permissions. Granting root privileges can potentially be dangerous, as it can give a user unrestricted access to the underlying operating system.

File: contrib/bentoml/README.md
Comment (ssheng) [reply contrib/bentoml/README.md:46]: Since this documentation is going to be used for 1.7, would you still like me to update the naming from Notebook to Workbench?

**Comment by ssheng** [comment]
> Plese sign CLA

Done. `cla/google` check is now passing. Please review again.

File: contrib/bentoml/README.md
Comment (juliusvonkohout) [reply contrib/bentoml/README.md:46]: i would prefer it. @DomFleischmann @kimwnasptd  what do you think?

Comment (juliusvonkohout) [reply contrib/bentoml/README.md:46]: @jbottum

**Comment by ssheng** [comment]
@juliusvonkohout Is there anything pending on our side? Please let us know.

**Comment by juliusvonkohout** [comment]
> @juliusvonkohout Is there anything pending on our side? Please let us know.

Sadly we have for @kimwnasptd to run the workflows and approve. I am not yet maintainer of the manifests repository.

File: contrib/bentoml/README.md
Comment (jbottum) [reply contrib/bentoml/README.md:46]: I would suggest using Notebooks for now.

**Comment by jbottum** [comment]
one note - when I view the file, I see 1.7.0-rc versions of components.    should those be 1.7.0, and not RC versions?

**Comment by ssheng** [comment]
> one note - when I view the file, I see 1.7.0-rc versions of components. should those be 1.7.0, and not RC versions?

@jbottum I'm happy to update it to `1.7.0`. I put `1.7.0-rc` because I saw other components were also referencing `1.7.0-rc`. I assume they will be updated together in a separate PR later.

**Comment by ssheng** [comment]
Changed upstream to 1.7.0 and fixed a test issue. May I please ask for another `/lgtm`, @jbottum?

**Comment by ssheng** [comment]
Hi, @elikatsis, @kimwnasptd, @PatrickXYS, @StefanoFioravanzo, @yanniszark, @annajung, could you please review when you get a chance?

File: contrib/bentoml/test.sh
Comment (ssheng) [inline contrib/bentoml/test.sh:34]:
```suggestion
  'http://localhost:3333/is_fraud' \
```

**Comment by juliusvonkohout** [comment]
@ssheng after that https://github.com/kubeflow/manifests/pull/2371 i can approve

File: contrib/bentoml/README.md
Comment (annajung) [inline contrib/bentoml/README.md:7]: Is this not supported with 1.25? 1.7 release supports both 1.24/1.25

File: README.md
Comment (annajung) [inline README.md:305]:
```suggestion
kustomize build contrib/bentoml/bentoml-yatai-stack/default | kubectl apply -n kubeflow --server-side -f -
```

File: example/kustomization.yaml
Comment (annajung) [inline example/kustomization.yaml:87]: This is not a blocker comment, I would like to understand how a third-party component is enabled by default vs. not. What makes bento qualified to be installed by default when other contrib components (other than kserve) are not?

Kserve is a special case as they were developed as part of Kubeflow before graduating out of Kubeflow project. Defining a process here would make it fair for all other contrib components.

File: contrib/bentoml/deployment_from_bento.yaml
Comment (annajung) [inline contrib/bentoml/deployment_from_bento.yaml:13]: nit, not a blocker. does it need three runners? kubeflow itself is pretty resource heavy so wondering if lightweight would be better

File: example/kustomization.yaml
Comment (juliusvonkohout) [reply example/kustomization.yaml:87]: Yes, that has to be removed for the time being. We will not install it by default.

File: contrib/bentoml/README.md
Comment (ssheng) [reply contrib/bentoml/README.md:7]: 1.25 is supported. Updated the doc.

File: example/kustomization.yaml
Comment (ssheng) [reply example/kustomization.yaml:87]: Thanks for the clarification. Removed.

File: contrib/bentoml/deployment_from_bento.yaml
Comment (ssheng) [reply contrib/bentoml/deployment_from_bento.yaml:13]: Using multiple runners is part of the demonstration of BentoML capabilities. Each pod only requests 0.1 CPU cores so hopefully resource is not a concern.

File: README.md
Comment (annajung) [inline README.md:61]: Can we remove this as well as it's not official component of Kubeflow and is not installed by default

File: README.md
Comment (annajung) [inline README.md:298]: Also remove this section as we don't list contrib components in the readme

File: contrib/bentoml/README.md
Comment (annajung) [inline contrib/bentoml/README.md:7]: Could you also update https://github.com/kubeflow/manifests/blob/master/.github/workflows/bentoml_kind_test.yaml to ensure it's tested with 1.25 cluster?

File: contrib/bentoml/OWNERS
Comment (annajung) [inline contrib/bentoml/OWNERS:3]: Based on [contrib guideline](https://github.com/kubeflow/manifests/blob/master/proposals/20220926-contrib-component-guidelines.md), it's correct that you need two owners, but contrib owners are the one who is responsible for the content in this directory that can make changes on behalf of BentoML.

Would it be more appropriate it add yourself here?

**Review by annajung — CHANGES_REQUESTED** [review]

**Comment by annajung** [comment]
cc @kimwnasptd looks like [github actions are not running and require approval from a maintainer](https://github.com/kubeflow/manifests/actions/runs/4611595124), could you help trigger the workflow?

File: contrib/bentoml/README.md
Comment (ssheng) [reply contrib/bentoml/README.md:7]: Is there an example that has already been updated to 1.25, which I can follow?

File: contrib/bentoml/OWNERS
Comment (ssheng) [reply contrib/bentoml/OWNERS:3]: Happy to add myself to the list.

File: README.md
Comment (ssheng) [reply README.md]: Removed

File: README.md
Comment (ssheng) [reply README.md]: Removed.

**Comment by annajung** [comment]
I didn't realize that you were not part of the Kubeflow org. Since non members can't be added to the OWNERS file, I suggest removing yourself for now and follow the [process](https://github.com/kubeflow/internal-acls#joining-kubeflow-github-organization) to join the Kubeflow org. We can get you added as an owner after you have joined the org.

once the OWNERS file is fixed and the workflow passes, it looks good to me. 

Thanks for all your work in updating the bentoML docs for the community.

**Comment by ssheng** [comment]
> I didn't realize that you were not part of the Kubeflow org. Since non members can't be added to the OWNERS file, I suggest removing yourself for now and follow the [process](https://github.com/kubeflow/internal-acls#joining-kubeflow-github-organization) to join the Kubeflow org. We can get you added as an owner after you have joined the org.
> 
> once the OWNERS file is fixed and the workflow passes, it looks good to me.
> 
> Thanks for all your work in updating the bentoML docs for the community.

Thanks for your revidw, Anna. Just removed myself from the OWNERS file. I will follow the process and join the Kubefow org.

**Pull request merged**
